src/interface/swarm_comms.cpp

```diff
@@ -250,18 +250,20 @@ void Swarm::LoadBuffers_() {
           auto p_index = buffer_sorted(n).swarm_idx_;
           if (swarm_d.IsActive(p_index)) {
             const int m = buffer_sorted(n).sort_idx_;
-            const int bufid = neighbor_buffer_index(m);
             if (m >= 0) {
-              const int bid = n - buffer_start[m];
-              int buffer_index = bid * particle_size;
-              swarm_d.MarkParticleForRemoval(p_index);
-              for (int i = 0; i < realPackDim; i++) {
-                bdvar.send[bufid](buffer_index) = vreal(i, p_index);
-                buffer_index++;
-              }
-              for (int i = 0; i < intPackDim; i++) {
-                bdvar.send[bufid](buffer_index) = static_cast<Real>(vint(i, p_index));
-                buffer_index++;
+              const int bufid = neighbor_buffer_index(m);
+              if (m >= 0) {
```

Comment:
Is this `m >=0` duplicated with line 253 above? Is that intentional?

Reply:
Thanks @Yurlungur, not sure how I missed that, fixed. :+1: